app/models/api_key.rb

```diff
@@ -67,6 +69,24 @@ def request_allowed?(env)
 
     api_key_scopes.blank? || api_key_scopes.any? { |s| s.permits?(env) }
   end
+
+  def last_seen_redis_key(now)
+    now_date = now.to_date
+    "api_key:#{id}:last_seen:#{now_date}"
+  end
+
+  def update_last_used!(now = Time.zone.now)
+    redis_key = last_seen_redis_key(now)
+
+    return if !Discourse.redis.set(
```

Comment:
Why reach out to redis here? We've already loaded the `last_seen` value from postgres, so couldn't we could just say

```
return if `last_used > 1.minute.ago`
```